src/ui/server/index.ts

```diff
@@ -166,4 +179,13 @@ ui.get('*', (req: Request, res: Response) => {
   });
 });
 
-https(ui);
+/**
+ * When being executed on Azure only HTTP server should be running,
+ * since TLS will be stripped. However ensure localhost and GHA
+ * are executed on TLS.
+ */
+if (isProduction()) {
+  http(ui);
+} else {
+  https(ui);
+}
```

Comment:
Really not a fan of environment specific logic in a codebase. I think "if test/dev environment" is ok, but "if production" is risker. Not a blocker, understand we have to do this for the Azure restriction(s).

Could we add a specific comment/description about the restriction, e.g "Azure handles https so we have to execute as http inside a wrapper", instead of "only http server..."? Just if you skim read this it feels strange